sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -2008,6 +2014,32 @@ class AdaptiveQueryExecSuite
       }
     }
   }
+
+  test("SPARK-36424: Support eliminate limits in AQE Optimizer") {
+    withTempView("v") {
+      spark.sparkContext.parallelize(
+        (1 to 10).map(i => TestData(i, if (i > 2) "2" else i.toString)), 2)
```

Comment:
we can use `spark.range` if you just want to control the number of partitions.

Reply:
Actually I don't like use range here since it has output partitionging. If you don't mind I prefer use `parallelize`.